xdsl/dialects/gpu.py

```diff
@@ -274,6 +274,37 @@ def get(buffer: SSAValue | Operation,
             result_types=[[AsyncTokenType()] if is_async else []])
 
 
+@irdl_op_definition
+class MemcpyOp(Operation):
+    name = "gpu.memcpy"
+
+    asyncDependencies: Annotated[VarOperand, AsyncTokenType]
+    src: Annotated[Operand, memref.MemRefType]
+    dst: Annotated[Operand, memref.MemRefType]
+
+    irdl_options = [AttrSizedOperandSegments()]
+
+    asyncToken: Annotated[OptOpResult, AsyncTokenType]
+
+    @staticmethod
+    def get(source: SSAValue | Operation,
+            destination: SSAValue | Operation,
+            async_dependencies: Sequence[SSAValue | Operation] | None = None,
+            is_async: bool = False) -> MemcpyOp:
+        return MemcpyOp.build(
+            operands=[
+                async_dependencies if async_dependencies is not None else [],
```

Comment:
```suggestion
                async_dependencies if async_dependencies else [],
```

Reply:
Your suggestion is equivalent to `async_dependencies or []`, which has the "risk" that that it triggers on `async_dependencies == []. Oviously that wouldn't be a problem here, but we agreed somewhere else that we should be careful with this. We probably want to start a broader discussion on this and record a decision somewhere?

Reply:
Started the topic there https://xdsl.zulipchat.com/#narrow/stream/368604-Framework/topic/inline.20conditionals, marking this as resolved meanwhile!